core/server/models/base/when.js

```diff
@@ -0,0 +1,59 @@
+var config = require('../../config'),
```

Comment:
The basic idea of `when` is to avoid listening on events all over the server.
I would love to create a centralized place, where we listen on all kinds of events and then delegate.
But for now, we could also move this logic into `scheduling/post-scheduling`.